lib/synapse/service_watcher/zookeeper.rb

```diff
@@ -36,18 +36,22 @@ def start
       @zk = nil
 
       log.info "synapse: starting ZK watcher #{@name} @ hosts: #{@zk_hosts}, path: #{@discovery['path']}"
+      statsd.increment('synapse.zk_watcher.started')
       zk_connect
     end
 
     def stop
       log.warn "synapse: zookeeper watcher exiting"
+      statsd.increment('synapse.zk_watcher.clean_stop')
```

Comment:
yes, `abort`

Reply:
the dogstatsd way of doing this would be to something like `statsd.increment('synapse.zk_watcher.stop', :tags => ["clean:true"])` and then use `["clean:false"]` in the other case. this lets you more easily slice the metrics later on. in general, it might be helpful to look through this PR with an eye for where using tags would be appropriate

Reply:
I like that. Will change

Reply:
this seems to also be the wrong place to put this metric.  I'll move it to `synapse.rb -> def run -> ensure`, which represents the proper clean stop place. That will also cover all watchers.